test/test_dereference.py

```diff
@@ -221,9 +225,10 @@ class Container(MongoModel):
         # access through raw dicts not through __get__ of the field
         # cause __get__ can perform a query to db for reference fields
         # to dereference them using dereference_id function
-        self.assertEqual(
-            container._data['lst'][0]._data['ref']['name'],
-            'Aaron')
+        # Not sure how this check translates now that we have the LazyDecoder.
+        # self.assertEqual(
+        #     container._data['lst'][0]._data['ref']['name'],
+        #     'Aaron')
```

Comment:
Couldn't this check be written using `with no_auto_dereference` to prevent `__get__` from performing any queries?

Reply:
Good idea! I have updated the test case as appropriate.